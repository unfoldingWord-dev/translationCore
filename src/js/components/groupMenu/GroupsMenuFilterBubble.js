import React from 'react';
import PropTypes from 'prop-types';
import { Glyphicon } from 'react-bootstrap';

class GroupsMenuFilterBubble extends React.Component {

  handleBubbleRemove() {
    this.props.setFilter(this.props.name, false);
  }

  render() {
    return (
      <span className="filter-bubble-wrapper">
        <span className="filter-bubble">
          <Glyphicon className='filter-remove' glyph='remove' onClick={this.handleBubbleRemove.bind(this)} />
          <span className="filter-text">{this.props.text}</span>
        </span>
      </span>
    );
  }
}

GroupsMenuFilterBubble.propTypes = {
  name: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  setFilter: PropTypes.func.isRequired
};

export default GroupsMenuFilterBubble;
