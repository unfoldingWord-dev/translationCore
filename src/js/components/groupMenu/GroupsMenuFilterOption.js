import React from 'react';
import PropTypes from 'prop-types';

class GroupsMenuFilterOption extends React.Component {

  handleCheckboxChange(event) {
    const value = event.target.checked;
    this.props.setFilter(this.props.name, value);
  }

  render() {
    const {
      name,
      text,
      icon,
      checked,
      disabled
    } = this.props;
    return (
      <label className={"option"+(disabled?" disabled":"")}>
        <span className="option-checkbox">
          <input type="checkbox" name={name} checked={checked} disabled={disabled} onChange={this.handleCheckboxChange.bind(this)} />
        </span>
        <span className="option-icon">
          {icon}
        </span>
        <span className="option-text">
          {text}
        </span>
      </label>
    );
  }
}

GroupsMenuFilterOption.defaultProps = {
  checked: false,
  disabled: false
};

GroupsMenuFilterOption.propTypes = {
  name: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  icon: PropTypes.object.isRequired,
  setFilter: PropTypes.func.isRequired,
  checked: PropTypes.bool,
  disabled: PropTypes.bool
};

export default GroupsMenuFilterOption;
