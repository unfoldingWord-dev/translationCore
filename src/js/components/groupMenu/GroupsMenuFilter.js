import React from 'react';
import PropTypes from 'prop-types';
import { Glyphicon } from 'react-bootstrap';
import InvalidatedIcon from '../svgIcons/InvalidatedIcon';
import GroupsMenuFilterOption from './GroupsMenuFilterOption';
import GroupsMenuFilterBubble from './GroupsMenuFilterBubble';

class GroupsMenuFilter extends React.Component {

  expandedFilter() {
    const {
      filters,
      translate
    } = this.props;
    const options = [];

    options.push(<GroupsMenuFilterOption
      key="invalidated"
      name="invalidated"
      checked={filters.invalidated}
      setFilter={this.props.setFilter}
      icon={<InvalidatedIcon width={16} height={16} color="#fff" />}
      text={translate('tools.invalidated')}/>);

    options.push(<GroupsMenuFilterOption
      key="reminders"
      name="reminders"
      checked={filters.reminders}
      setFilter={this.props.setFilter}
      icon={<Glyphicon glyph="bookmark" />}
      text={translate('tools.bookmarks')}/>);

    options.push(<GroupsMenuFilterOption
      key="selections"
      name="selections"
      checked={filters.selections}
      disabled={filters.noSelections}
      setFilter={this.props.setFilter}
      icon={<Glyphicon glyph="ok" />}
      text={translate('tools.selected')}/>);

    options.push(<GroupsMenuFilterOption
      key="noSelections"
      name="noSelections"
      checked={filters.noSelections}
      disabled={filters.selections}
      setFilter={this.props.setFilter}
      icon={<Glyphicon glyph="ban-circle" />}
      text={translate('tools.no_selection')}/>);

    options.push(<GroupsMenuFilterOption
      key="verseEdits"
      name="verseEdits"
      checked={filters.verseEdits}
      setFilter={this.props.setFilter}
      icon={<Glyphicon glyph="pencil" />}
      text={translate('tools.verse_edit')}/>);

    options.push(<GroupsMenuFilterOption
      key="comments"
      name="comments"
      checked={filters.comments}
      setFilter={this.props.setFilter}
      icon={<Glyphicon glyph="comment" />}
      text={translate('tools.comments')}/>);

    return (
      <div id="groups-menu-filter" className="options-wrapper">
        {options}
      </div>
    );
  }

  collapsedFilter() {
    const {
      translate,
      filters
    } = this.props;
    const bubbles = [];

    if (filters.invalidated) {
      bubbles.push(<GroupsMenuFilterBubble
        key='invalidated'
        name='invalidated'
        text={translate('tools.invalidated')}
        setFilter={this.props.setFilter} />);
    }

    if (filters.reminders) {
      bubbles.push(<GroupsMenuFilterBubble
        key='reminders'
        name='reminders'
        text={translate('tools.bookmarks')}
        setFilter={this.props.setFilter} />);
    }

    if (filters.selections) {
      bubbles.push(<GroupsMenuFilterBubble
        key='selections'
        name='selections'
        text={translate('tools.selected')}
        setFilter={this.props.setFilter} />);
    }

    if (filters.noSelections) {
      bubbles.push(<GroupsMenuFilterBubble
        key='noSelections'
        name='noSelections'
        text={translate('tools.no_selection')}
        setFilter={this.props.setFilter} />);
    }

    if (filters.verseEdits) {
      bubbles.push(<GroupsMenuFilterBubble
        key='verseEdits'
        name='verseEdits'
        text={translate('tools.verse_edit')}
        setFilter={this.props.setFilter} />);
    }
    
    if (filters.comments) {
      bubbles.push(<GroupsMenuFilterBubble
        key='comments'
        name='comments'
        text={translate('tools.comments')}
        setFilter={this.props.setFilter} />);
    }

    return (
      <div id="groups-menu-filter" className="bubbles-wrapper">
        {bubbles}
      </div>
    );
  }

  render() {
    if (this.props.expandFilter) {
      return this.expandedFilter();
    } else {
      return this.collapsedFilter();
    }
  }
}

GroupsMenuFilter.defaultProps = {
  expandFilter: false
};

GroupsMenuFilter.propTypes = {
  translate: PropTypes.func.isRequired,
  filters: PropTypes.object.isRequired,
  setFilter: PropTypes.func,
  expandFilter: PropTypes.bool
};

export default GroupsMenuFilter;
