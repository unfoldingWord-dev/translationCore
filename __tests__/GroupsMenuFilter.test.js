/* eslint-disable no-console */
/* eslint-env jest */
import React from 'react';
import GroupsMenuFilter from '../src/js/components/groupMenu/GroupsMenuFilter';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import GroupsMenuFilterBubble from '../src/js/components/groupMenu/GroupsMenuFilterBubble';
import GroupsMenuFilterOption from '../src/js/components/groupMenu/GroupsMenuFilterOption';

describe('Tests for GroupsMenuFilter', () => {
  test('Test GroupsMenuFilter for translationWords expanded', () => {
    // given
    const numOptions = 6;
    const numBubbles = 0;
    const numCheckedOptions = 6;

    const props = {
      currentToolName: 'translationWords',
      translate: k=>k,
      setFilter: jest.fn(),
      expandFilter: true,
      filters: {
        invalidated: true,
        reminders: true,
        selections: true,
        noSelections: true,
        verseEdits: true,
        comments: true
      }
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilter {...props} />
    );
    const filterOptions = wrapper.find(GroupsMenuFilterOption);
    const checkedOptions = wrapper.find({ checked: true })
    const filterBubbles = wrapper.find(GroupsMenuFilterBubble);

    // then
    expect(filterOptions.length).toEqual(numOptions);
    expect(filterBubbles.length).toEqual(numBubbles);
    expect(checkedOptions.length).toEqual(numCheckedOptions);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Test GroupsMenuFilter for translationWords expanded with 3 options checked', () => {
    // given
    const numOptions = 6;
    const numBubbles = 0;
    const numCheckedOptions = 3;

    const props = {
      currentToolName: 'translationWords',
      translate: k=>k,
      setFilter: jest.fn(),
      filters: {
        invalidated: false,
        reminders: true,
        selections: false,
        noSelections: true,
        verseEdits: false,
        comments: true
      },
      expandFilter: true
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilter {...props} />
    );
    const filterOptions = wrapper.find(GroupsMenuFilterOption);
    const checkedOptions = wrapper.find({ checked: true })
    const filterBubbles = wrapper.find(GroupsMenuFilterBubble);

    // then
    expect(filterOptions.length).toEqual(numOptions);
    expect(filterBubbles.length).toEqual(numBubbles);
    expect(checkedOptions.length).toEqual(numCheckedOptions);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Test GroupsMenuFilter for translationWords collapsed with all options', () => {
    // given
    const numOptions = 0;
    const numBubbles = 6;

    const props = {
      currentToolName: 'translationWords',
      translate: k=>k,
      setFilter: jest.fn(),
      filters: {
        invalidated: true,
        reminders: true,
        selections: true,
        noSelections: true,
        verseEdits: true,
        comments: true
      },
      expandFilter: false
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilter {...props} />
    );
    const filterOptions = wrapper.find(GroupsMenuFilterOption);
    const filterBubbles = wrapper.find(GroupsMenuFilterBubble);

    // then
    expect(filterOptions.length).toEqual(numOptions);
    expect(filterBubbles.length).toEqual(numBubbles);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Test GroupsMenuFilter for translationWords collapsed with no options', () => {
    // given
    const numOptions = 0;
    const numBubbles = 0;

    const props = {
      currentToolName: 'translationWords',
      translate: k=>k,
      setFilter: jest.fn(),
      filters: {
        invalidated: false,
        reminders: false,
        selections: false,
        noSelections: false,
        verseEdits: false,
        comments: false
      },
      expandFilter: false
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilter {...props} />
    );
    const filterOptions = wrapper.find(GroupsMenuFilterOption);
    const filterBubbles = wrapper.find(GroupsMenuFilterBubble);

    // then
    expect(filterOptions.length).toEqual(numOptions);
    expect(filterBubbles.length).toEqual(numBubbles);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
