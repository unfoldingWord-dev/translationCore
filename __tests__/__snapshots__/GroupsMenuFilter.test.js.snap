// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GroupsMenuFilter Test GroupsMenuFilter for translationWords collapsed with all options 1`] = `
<div
  className="bubbles-wrapper"
  id="groups-menu-filter"
>
  <GroupsMenuFilterBubble
    key="invalidated"
    name="invalidated"
    setFilter={[Function]}
    text="tools.invalidated"
  />
  <GroupsMenuFilterBubble
    key="reminders"
    name="reminders"
    setFilter={[Function]}
    text="tools.bookmarks"
  />
  <GroupsMenuFilterBubble
    key="selections"
    name="selections"
    setFilter={[Function]}
    text="tools.selected"
  />
  <GroupsMenuFilterBubble
    key="noSelections"
    name="noSelections"
    setFilter={[Function]}
    text="tools.no_selection"
  />
  <GroupsMenuFilterBubble
    key="verseEdits"
    name="verseEdits"
    setFilter={[Function]}
    text="tools.verse_edit"
  />
  <GroupsMenuFilterBubble
    key="comments"
    name="comments"
    setFilter={[Function]}
    text="tools.comments"
  />
</div>
`;

exports[`Tests for GroupsMenuFilter Test GroupsMenuFilter for translationWords collapsed with no options 1`] = `
<div
  className="bubbles-wrapper"
  id="groups-menu-filter"
/>
`;

exports[`Tests for GroupsMenuFilter Test GroupsMenuFilter for translationWords expanded 1`] = `
<div
  className="options-wrapper"
  id="groups-menu-filter"
>
  <GroupsMenuFilterOption
    checked={true}
    disabled={false}
    icon={
      <InvalidatedIcon
        color="#fff"
        height={16}
        width={16}
      />
    }
    key="invalidated"
    name="invalidated"
    setFilter={[Function]}
    text="tools.invalidated"
  />
  <GroupsMenuFilterOption
    checked={true}
    disabled={false}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="bookmark"
      />
    }
    key="reminders"
    name="reminders"
    setFilter={[Function]}
    text="tools.bookmarks"
  />
  <GroupsMenuFilterOption
    checked={true}
    disabled={true}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="ok"
      />
    }
    key="selections"
    name="selections"
    setFilter={[Function]}
    text="tools.selected"
  />
  <GroupsMenuFilterOption
    checked={true}
    disabled={true}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="ban-circle"
      />
    }
    key="noSelections"
    name="noSelections"
    setFilter={[Function]}
    text="tools.no_selection"
  />
  <GroupsMenuFilterOption
    checked={true}
    disabled={false}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="pencil"
      />
    }
    key="verseEdits"
    name="verseEdits"
    setFilter={[Function]}
    text="tools.verse_edit"
  />
  <GroupsMenuFilterOption
    checked={true}
    disabled={false}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="comment"
      />
    }
    key="comments"
    name="comments"
    setFilter={[Function]}
    text="tools.comments"
  />
</div>
`;

exports[`Tests for GroupsMenuFilter Test GroupsMenuFilter for translationWords expanded with 3 options checked 1`] = `
<div
  className="options-wrapper"
  id="groups-menu-filter"
>
  <GroupsMenuFilterOption
    checked={false}
    disabled={false}
    icon={
      <InvalidatedIcon
        color="#fff"
        height={16}
        width={16}
      />
    }
    key="invalidated"
    name="invalidated"
    setFilter={[Function]}
    text="tools.invalidated"
  />
  <GroupsMenuFilterOption
    checked={true}
    disabled={false}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="bookmark"
      />
    }
    key="reminders"
    name="reminders"
    setFilter={[Function]}
    text="tools.bookmarks"
  />
  <GroupsMenuFilterOption
    checked={false}
    disabled={true}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="ok"
      />
    }
    key="selections"
    name="selections"
    setFilter={[Function]}
    text="tools.selected"
  />
  <GroupsMenuFilterOption
    checked={true}
    disabled={false}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="ban-circle"
      />
    }
    key="noSelections"
    name="noSelections"
    setFilter={[Function]}
    text="tools.no_selection"
  />
  <GroupsMenuFilterOption
    checked={false}
    disabled={false}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="pencil"
      />
    }
    key="verseEdits"
    name="verseEdits"
    setFilter={[Function]}
    text="tools.verse_edit"
  />
  <GroupsMenuFilterOption
    checked={true}
    disabled={false}
    icon={
      <Glyphicon
        bsClass="glyphicon"
        glyph="comment"
      />
    }
    key="comments"
    name="comments"
    setFilter={[Function]}
    text="tools.comments"
  />
</div>
`;
